DevOps.Util/BaseServer.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DevOps.Util
+{
+    public class BaseServer
+    {
+        public AuthorizationToken AuthorizationToken { get; }
+        public HttpClient HttpClient { get; }
+
+        public BaseServer(AuthorizationToken authorizationToken = default, HttpClient? httpClient = null)
+        {
+            AuthorizationToken = authorizationToken;
+            HttpClient = httpClient ?? new HttpClient();
+        }
+
+        public async Task DownloadFileAsync(string uri, Stream destinationStream)
```

Comment:
Okay I think i see the intent now. Rather than inheritance on the server though I think it would be cleaner to provide an abstraction around `HttpClient`. That is the actual behavior that is being modified here.